specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/stable/2020-03-01/outputs.json

```diff
@@ -0,0 +1,1066 @@
+        },
+        "propertyColumns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "AzureSqlDatabaseOutputDataSource": {
+      "description": "Describes an Azure SQL database output data source.",
+      "x-ms-discriminator-value": "Microsoft.Sql/Server/Database",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OutputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AzureSqlDatabaseOutputDataSourceProperties",
+          "description": "The properties that are associated with an Azure SQL database output. Required on PUT (CreateOrReplace) requests.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "AzureSqlDatabaseOutputDataSourceProperties": {
+      "description": "The properties that are associated with an Azure SQL database output.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureSqlDatabaseDataSourceProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "AzureSynapseOutputDataSource": {
+      "description": "Describes an Azure Synapse output data source.",
+      "x-ms-discriminator-value": "Microsoft.Sql/Server/DataWarehouse",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OutputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AzureSynapseOutputDataSourceProperties",
+          "description": "The properties that are associated with an Azure Synapse output. Required on PUT (CreateOrReplace) requests.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "AzureSynapseOutputDataSourceProperties": {
+      "description": "The properties that are associated with an Azure Synapse output.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureSynapseDataSourceProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "AzureSynapseDataSourceProperties": {
+      "description": "The properties that are associated with an Azure SQL database data source.",
+      "properties": {
+        "server": {
+          "type": "string",
+          "description": "The name of the SQL server containing the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "database": {
+          "type": "string",
+          "description": "The name of the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "table": {
+          "type": "string",
+          "description": "The name of the table in the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "user": {
+          "type": "string",
+          "description": "The user name that will be used to connect to the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "password": {
```

Comment:
>password [](http://example.com/codeflow?start=9&length=8)

There is a bigger issue with this property. Secrets are not allowed in PUT, PATCH or GET responses, but this type is part of the "Output" definition that is present in all of those payloads. This makes it doubly important you tag it with the x-ms-secret extension.

Reply:
We made the decision a long time ago to only have one definition for all the requests to make the SDK experience easier for customers. Our workaround was to add more information in the description such as "Required on PUT (CreateOrReplace) requests" to indicate that this property is required on PUT requests, but not PATCH requests, for example. Can we keep it like this to maintain consistency?

Reply:
Removing it from the response would break existing clients, so I won't block approval on it. Your RP should at least blank or mask it from GET, PUT, and PATCH responses. If there's a need to retrieve it, you should only return it in the POST action.

Reply:
Yes, we made the conscious decision to only use one definition for all of our apis to make it more intuitive for customers. So in this case, we only have one Output definition, instead of having a PatchOutput definition, and another for GetOutput, etc. I would prefer to leave it like this to maintain consistency.

Reply:
If I understand the original comment, the concern is that secrets are returned the **response** payload of PUT, PATCH, or GET responses. I can guarantee we are following that. Although it may be supplied as part of the request payload, we remove secret values from the response before sending it back to the customer (which has garnered a number of customer complaints for us btw :) although I understand the security aspect of the requirement)